portfolio/src/test/java/com/google/sps/TimeRangeGroupTest.java

```diff
@@ -0,0 +1,669 @@
+    Instant timeRangeCombinedEnd = timeRangeToAddEnd;
+    TimeRange timeRangeNew = TimeRange.fromStartEnd(timeRangeCombinedStart, timeRangeCombinedEnd);
+    List<TimeRange> expected = Arrays.asList(timeRangeTwo, timeRangeNew);
+
+    timeRangeGroup.addTimeRange(timeRangeToAdd);
+    Iterator<TimeRange> actualIterator = timeRangeGroup.iterator();
+
+    List<TimeRange> actual = new LinkedList();
+    while (actualIterator.hasNext()) {
+      actual.add(actualIterator.next());
+    }
+
+    Collections.sort(actual, TimeRange.SORT_BY_TIME_RANGE_DURATION_ASCENDING_THEN_START_TIME);
+    Assert.assertEquals(expected, actual);
+  }
+
+  /** Tests for adding a time range that overlaps another existing one. */
+  @Test
+  public void testAddOverlappingRangeTwo() {
+    // Time Ranges: |-----A-----|     |----C----|
+    // To add:                            |-----B-----|
+    // Result:      |-----------|     |---------------|
+    Instant timeRangeOneStart = Instant.now();
+    Instant timeRangeOneEnd = timeRangeOneStart.plusSeconds(1000);
+    Instant timeRangeTwoStart = timeRangeOneEnd.plusSeconds(3000);
+    Instant timeRangeTwoEnd = timeRangeTwoStart.plusSeconds(1000);
+
+    TimeRange timeRangeOne = TimeRange.fromStartEnd(timeRangeOneStart, timeRangeOneEnd);
+    TimeRange timeRangeTwo = TimeRange.fromStartEnd(timeRangeTwoStart, timeRangeTwoEnd);
+
+    List<TimeRange> originalTimeRanges = Arrays.asList(timeRangeTwo, timeRangeOne);
+    for (TimeRange timeRange : originalTimeRanges) {
+      timeRangeGroup.addTimeRange(timeRange);
+    }
+
+    Instant timeRangeToAddStart = timeRangeTwoStart.plusSeconds(500);
+    Instant timeRangeToAddEnd = timeRangeTwoEnd.plusSeconds(1000);
+    TimeRange timeRangeToAdd = TimeRange.fromStartEnd(timeRangeToAddStart, timeRangeToAddEnd);
+
+    // Constructs the expected merged time range for combining B and C.
+    Instant timeRangeCombinedStart = timeRangeTwoStart;
+    Instant timeRangeCombinedEnd = timeRangeToAddEnd;
+    TimeRange timeRangeNew = TimeRange.fromStartEnd(timeRangeCombinedStart, timeRangeCombinedEnd);
+    List<TimeRange> expected = Arrays.asList(timeRangeOne, timeRangeNew);
+
+    timeRangeGroup.addTimeRange(timeRangeToAdd);
+    Iterator<TimeRange> actualIterator = timeRangeGroup.iterator();
+
+    List<TimeRange> actual = new LinkedList();
+    while (actualIterator.hasNext()) {
+      actual.add(actualIterator.next());
+    }
+
+    Collections.sort(actual, TimeRange.SORT_BY_TIME_RANGE_DURATION_ASCENDING_THEN_START_TIME);
+    Assert.assertEquals(expected, actual);
+  }
+
+  /** Tests for adding a time range that entirely contains an existing one. */
+  @Test
+  public void testAddSupersetRange() {
+    // Time Ranges:      |-A-|
+    // To add:       |-----B-----|
+    // Result:       |-----------|
+    Instant timeRangeOneStart = Instant.now();
+    Instant timeRangeOneEnd = timeRangeOneStart.plusSeconds(3000);
+    Instant timeRangeTwoStart = timeRangeOneStart.plusSeconds(500);
```

Comment:
timeRangeTwoStart should probably be `timeRangeOneStart.minusSeconds(500)` ?

Reply:
Fixed.